File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/Array_List/DoublePointer/Array/removeDuplicates_26.java
package algo.tzashinorpu.SpecialSubject.Array_List.DoublePointer.Array;

public class removeDuplicates_26 {
    public int removeDuplicates(int[] nums) {
        int l = 0;
        for (int r = l; r < nums.length; r++) {
            if (l != r && nums[r] != nums[l]) {
                nums[++l] = nums[r];
            }
        }
        return l;
    }
}

File: Clion/ZeroTrac/array_list/removeDuplicates_26.cpp
//
// Created by TzashiNorpu on 10/18/2022.
//
#include "array_list.h"

using namespace Array_List;

int Solution::removeDuplicates(vector<int> &nums) {
  int index = 0;
  for (int i = 1; i < nums.size(); ++i) {
    if (nums[index] != nums[i]) nums[++index] = nums[i];
  }
  return index + 1;
}